llvm/lib/CGData/StableFunctionMap.cpp

```diff
@@ -35,21 +36,30 @@ static cl::opt<unsigned> GlobalMergingMaxParams(
     cl::desc(
         "The maximum number of parameters allowed when merging functions."),
     cl::init(std::numeric_limits<unsigned>::max()), cl::Hidden);
-static cl::opt<unsigned> GlobalMergingParamOverhead(
+static cl::opt<bool> GlobalMergingSkipNoParams(
+    "global-merging-skip-no-params",
+    cl::desc("Skip merging functions with no parameters."), cl::init(false),
+    cl::Hidden);
+static cl::opt<double> GlobalMergingInstOverhead(
+    "global-merging-inst-overhead",
+    cl::desc("The overhead cost associated with each instruction when lowering "
+             "to machine instruction."),
+    cl::init(1.0), cl::Hidden);
```

Comment:
Just did a test with `1.2` and it delivered another 0.6% code size reduction on the same project I tested the effect of accurate `ParamCount`.

Reply:
I don't mind changing the default value as it could eventually change depending on target/app. In theory, it's hard to precisely estimate final size impact for the given IR, anyhow,